### PR TITLE
adding vowpal wabbit recipe

### PR DESCRIPTION
**Pull request by gramhagen**
Checklist

- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [X] Package does not vend other packages
- [X] Build number is 0
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

- note: I will need to update the source to pull from Vowpal Wabbit github once confirmed this tarball version works for win/linux/osx platforms

### CODE REVIEW

File: recipes/vowpalwabbit/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+test:
+  imports:
+    - vowpalwabbit
+  requires:
+    - numpy
+    - pandas
+    - pytest
+    - scipy
+    - scikit-learn
```
Comment (scopatz) [inline recipes/vowpalwabbit/meta.yaml]: Are none of these strict install requirements?

File: recipes/vowpalwabbit/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  sha256: befed5ecd42decd533b8e11a98f0e76f9d48951c085b52786277d86ec06923ac
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake>=3.13
```
Comment (scopatz) [inline recipes/vowpalwabbit/meta.yaml]: There needs to be a single space after the package name, e.g. `cmake >=3.13`

File: recipes/vowpalwabbit/meta.yaml (hunk 1 above)
Comment (gramhagen) [reply recipes/vowpalwabbit/meta.yaml]: they aren't strict requirements, one can use the package without them. however there is a sklearn wrapper included which does require sklearn and scipy, and the tests for this wrapper involve pandas

File: recipes/vowpalwabbit/meta.yaml (hunk 2 above)
Comment (gramhagen) [reply recipes/vowpalwabbit/meta.yaml]: got it

File: recipes/vowpalwabbit/meta.yaml (hunk 1 above)
Comment (scopatz) [reply recipes/vowpalwabbit/meta.yaml]: OK!

**Comment by gramhagen** [comment]
hmm, checking the conda-forge-linter issue I ran
```conda smithy recipe-lint .``` 
and got:
```jinja2.exceptions.UndefinedError: 'load_setup_py_data' is undefined```
but that is how this [page](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#jinja-templates) indicates you should pull the version from setup.py
Am I missing something?

**Comment by scopatz** [comment]
> Am I missing something?

That isn't something that is used by conda-forge.  That is more for making development packages.

**Comment by gramhagen** [comment]
> > Am I missing something?
> 
> That isn't something that is used by conda-forge. That is more for making development packages.

Oh I see, is there a preferred way to pull the version from the source code for conda-forge?

**Comment by scopatz** [comment]
> Oh I see, is there a preferred way to pull the version from the source code for conda-forge?

Generally we don't. Versions are hard-coded in the meta.yaml and then bumped and managed by bots once this is in a feedstock.  Having it pull versions from source code would almost certainly break the update bots.

File: recipes/vowpalwabbit/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  license_family: BSD
+  license_file: LICENSE
+  summary: Fast online machine learning library
+  doc_url: https://github.com/VowpalWabbit/vowpal_wabbit/wiki
+  dev_url: https://github.com/VowpalWabbit/vowpal_wabbit
+
+extra:
+  recipe-maintainers:
+    - gramhagen
+    - peterychang
```
Comment (scopatz) [inline recipes/vowpalwabbit/meta.yaml]: @peterychang are you OK with being a maintainer here?

Comment (peterychang) [reply recipes/vowpalwabbit/meta.yaml]: Yes, I'm ok with being a maintainer. Thanks!

File: recipes/vowpalwabbit/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    - scipy
+    - scikit-learn
+  source_files:
+    - python/tests
+  commands:
+    - py.test python/tests
+
+about:
+  home: https://github.com/VowpalWabbit/vowpal_wabbit
+  license: BSD
```
Comment (chrisburr) [inline recipes/vowpalwabbit/meta.yaml]: Which BSD licence is it? From a quick glance it looks like this but please check:
```suggestion
  license: BSD-3-Clause
```

File: recipes/vowpalwabbit/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
+package:
+  name: vowpalwabbit
+  version: 8.7.0
+
+source:
+  url: https://scgrahamarchive.blob.core.windows.net/vowpal-wabbit/vowpal_wabbit_2ee2bcf_rapidjson.tgz
```
Comment (chrisburr) [inline recipes/vowpalwabbit/meta.yaml]: Reminder to replace this with the upstream tarball (applying patches if required)

Comment (gramhagen) [reply recipes/vowpalwabbit/meta.yaml]: @peterychang when can we get the tarball on pypi? should we hold on this till then?

Comment (peterychang) [reply recipes/vowpalwabbit/meta.yaml]: We just discussed the plan for the official pypi release. We should have the source tarball up either today or tomorrow, so its probably best to wait until its done

File: recipes/vowpalwabbit/meta.yaml [hunk 6]
```diff
@@ -3,13 +3,13 @@ package:
   version: 8.7.0
 
 source:
-  url: https://scgrahamarchive.blob.core.windows.net/vowpal-wabbit/vowpal_wabbit_2ee2bcf_rapidjson.tgz 
-  sha256: befed5ecd42decd533b8e11a98f0e76f9d48951c085b52786277d86ec06923ac
+  url: https://files.pythonhosted.org/packages/2f/fb/1dc86a73579f892762aabebf6fca3c492f77a1f5a6d1035e6e6b6ead89e6/vowpalwabbit-8.7.0.tar.gz
```
Comment (chrisburr) [inline recipes/vowpalwabbit/meta.yaml]: Can you look at the example and change this with the predicable URL so the update bot can work?

Comment (gramhagen) [reply recipes/vowpalwabbit/meta.yaml]: fixed

**Comment by gramhagen** [comment]
ok, source was updated. I think we're good to go now, unless there are other code review items

**Pull request merged**